framework/src/utils/PetscSupport.C

```diff
@@ -354,6 +354,55 @@ petscNonlinearConverged(SNES /*snes*/,
   PetscFunctionReturn(PETSC_SUCCESS);
 }
 
+PetscErrorCode
+petscLinearConverged(
+    KSP /*ksp*/, PetscInt it, PetscReal /*norm*/, KSPConvergedReason * reason, void * ctx)
+{
+  PetscFunctionBegin;
+  FEProblemBase & problem = *static_cast<FEProblemBase *>(ctx);
+
+  // execute objects that may be used in convergence check
+  // TODO: add a dedicated flag once support is fully implemented.
+  // Right now, setting objects to execute on this flag would be ignored except in the
+  // linear-system-only use case.
+  problem.execute(EXEC_NONLINEAR_CONVERGENCE);
```

Comment:
add additional exec flag instead of using the `EXEC_LINEAR_CONVERGENCE`

Reply:
I assume you mean EXEC_NONLINEAR_CONVERGENCE.
I added EXEC_LINEAR_CONVERGENCE